resources/ts/Components/Integrations/Detail/Credentials.tsx

```diff
@@ -40,7 +41,11 @@ export const Credentials = ({
   const hasAnyCredentials = Boolean(
     legacyAuthConsumers.length || authClients.length
   );
-  usePolling(!hasAnyCredentials, { only: ["integration"] });
+  usePolling(
+    !hasAnyCredentials ||
+      (keyVisibility === KeyVisibility.v1 && !!keyVisibilityUpgrade),
```

Comment:
Minor but I would maybe but the second part of the condition in a variable like `hasAnyCredentials` was for readability purposes